Color Picker/ColorPickerScreen.swift

```diff
@@ -155,222 +282,83 @@ struct ColorPickerScreen: View {
-					if
-						isTextFieldFocusedRGB,
-						!isPreventingUpdate,
-						let newColor = NSColor(cssRGBString: $0.trimmingCharacters(in: .whitespaces))
-					{
-						colorPanel.color = newColor
-					}
-
-					if !isPreventingUpdate {
-						updateColorsFromPanel(excludeRGB: true, preventUpdate: true)
-					}
-				}
-			Button("Copy RGB", systemImage: "doc.on.doc.fill") {
-				rgbColor.copyToPasteboard()
-			}
-				.labelStyle(.iconOnly)
-				.symbolRenderingMode(.hierarchical)
-				.buttonStyle(.borderless)
-				.contentShape(.rectangle)
-				.keyboardShortcut("r", modifiers: [.shift, .command])
-		}
-	}
-
-	private var lchColorView: some View {
-		HStack {
-			NativeTextField(
-				text: $lchColor,
-				placeholder: "LCH",
-				font: .monospacedSystemFont(ofSize: textFieldFontSize, weight: .regular),
-				isFocused: $isTextFieldFocusedLCH
-			)
-				.controlSize(.large)
-				.onChange(of: lchColor) {
-					if
-						isTextFieldFocusedLCH,
-						!isPreventingUpdate,
-						let newColor = NSColor(cssLCHString: $0.trimmingCharacters(in: .whitespaces))
-					{
-						colorPanel.color = newColor
-					}
-
-					if !isPreventingUpdate {
-						updateColorsFromPanel(excludeLCH: true, preventUpdate: true)
-					}
-				}
-			Button("Copy LCH", systemImage: "doc.on.doc.fill") {
-				lchColor.copyToPasteboard()
-			}
-				.labelStyle(.iconOnly)
-				.symbolRenderingMode(.hierarchical)
-				.buttonStyle(.borderless)
-				.contentShape(.rectangle)
-				.keyboardShortcut("l", modifiers: [.shift, .command])
-		}
-	}
-
 	var body: some View {
 		VStack {
 			BarView()
 			if shownColorFormats.contains(.hex) {
-				hexColorView
+                ColorInputView(
+                    inputColorText: $appState.hexColor,
+                    isTextFieldFocused: $isTextFieldFocusedHex,
+                    colorPanel: colorPanel,
+                    textFieldFontSize: textFieldFontSize,
+                    inputColorType: .hex
+                )
 			}
 			if shownColorFormats.contains(.hsl) {
-				hslColorView
+                ColorInputView(
+                    inputColorText: $appState.hslColor,
+                    isTextFieldFocused: $isTextFieldFocusedHSL,
+                    colorPanel: colorPanel,
+                    textFieldFontSize: textFieldFontSize,
+                    inputColorType: .hsl
+                )
 			}
 			if shownColorFormats.contains(.rgb) {
-				rgbColorView
+                ColorInputView(
+                    inputColorText: $appState.rgbColor,
+                    isTextFieldFocused: $isTextFieldFocusedRGB,
+                    colorPanel: colorPanel,
+                    textFieldFontSize: textFieldFontSize,
+                    inputColorType: .rgb
+                )
 			}
 			if shownColorFormats.contains(.lch) {
-				lchColorView
+                ColorInputView(
+                    inputColorText: $appState.lchColor,
+                    isTextFieldFocused: $isTextFieldFocusedLCH,
+                    colorPanel: colorPanel,
```

Comment:
The color panel is available on the AppState, so you can access it from there instead.